File: problemas/1100/1179.dart
import 'dart:io';

void main() {
  final pares = new List<int>();
  final impares = new List<int>();

  for (var i = 0; i < 15; i++) {
    int n = int.parse(stdin.readLineSync());

    if (impares.length == 5) {
        imprimeNumeros(impares, false);
    }
    else if (pares.length == 5) {
        imprimeNumeros(pares, true);
    }

    if (n % 2 == 0) {
        pares.add(n);
    }
    else
    {
        impares.add(n);
    }
  }

  if (!impares.isEmpty) {
      imprimeNumeros(impares, false);
  }

  if (!pares.isEmpty) {
      imprimeNumeros(pares, true);
  }
}

void imprimeNumeros(List<int> numeros, bool saoPares) {
  for (int i = 0; i < numeros.length; i++) {
    stdout.writeln('${saoPares ? 'par' : 'impar'}[$i] = ${numeros[i]}');
  }

  numeros.clear();
}

File: problemas/1100/1179.py
def imprime_numeros(numeros, sao_pares):
    for i in range(len(numeros)):
        print('par' if sao_pares else 'impar', end='')
        print(f'[{i}] = {numeros[i]}')

    numeros.clear()


pares = []
impares = []

for j in range(15):
    n = int(input())

    if len(impares) == 5:
        imprime_numeros(impares, False)
    elif len(pares) == 5:
        imprime_numeros(pares, True)

    if n % 2 == 0:
        pares.append(n)
    else:
        impares.append(n)

if len(impares) > 0:
    imprime_numeros(impares, False)

if len(pares) > 0:
    imprime_numeros(pares, True)
